File: app/src/main/java/com/example/campusrent/Item.java
package com.example.campusrent;

import com.google.firebase.firestore.ServerTimestamp;

import java.util.Date;

public class Item {
    private String name;
    private String description;
    private double price;
    private String imageUrl;
    private String userId;
    private Date timestamp;

    public Item() {
        // Needed for Firestore
    }

    public Item(String name, String description, double price, String imageUrl, String userId) {
        this.name = name;
        this.description = description;
        this.price = price;
        this.imageUrl = imageUrl;
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public double getPrice() {
        return price;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getUserId() {
        return userId;
    }

    @ServerTimestamp
    public Date getTimestamp() {
        return timestamp;
    }
}

File: app/src/main/java/com/example/campusrent/data/model/Item.kt
package com.example.campusrent.data.model

import com.google.gson.annotations.SerializedName

data class Item(
    @SerializedName("id") val id: String = "",
    @SerializedName("name") val name: String = "",
    @SerializedName("price") val price: Double = 0.0,
    @SerializedName("description") val description: String = "",
    @SerializedName("image_url") val imageUrl: String = "",
    @SerializedName("user_id") val userId: String = ""
)
